### PR TITLE
remove support of hidden_envoy_deprecated_cluster

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: remove support of hidden_envoy_deprecated_cluster
Additional Description:
Risk Level: LOW
Testing: Unit
Docs Changes: NA
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @tyxia

File: source/extensions/filters/network/redis_proxy/config.cc [hunk 1]
```diff
@@ -47,11 +47,6 @@ Network::FilterFactoryCb RedisProxyFilterConfigFactory::createFilterFactoryFromP
   envoy::extensions::filters::network::redis_proxy::v3::RedisProxy::PrefixRoutes prefix_routes(
       proto_config.prefix_routes());
 
-  // Set the catch-all route from the deprecated cluster and settings parameters.
-  if (prefix_routes.routes_size() == 0 && !prefix_routes.has_catch_all_route()) {
```
Comment (htuch) [inline source/extensions/filters/network/redis_proxy/config.cc:51]: What does this have to do with the deprecated field?

File: test/extensions/filters/network/redis_proxy/config_test.cc [hunk 2]
```diff
@@ -34,9 +34,6 @@ TEST(RedisProxyFilterConfigFactoryTest, NoUpstreamDefined) {
 
   NiceMock<Server::Configuration::MockFactoryContext> context;
 
-  EXPECT_THROW_WITH_MESSAGE(
```
Comment (htuch) [inline test/extensions/filters/network/redis_proxy/config_test.cc:37]: Needs an EXPECT of some kind for the test.

**Review by htuch — CHANGES_REQUESTED** [review]
/wait

File: source/extensions/filters/network/redis_proxy/config.cc (hunk 1 above)
Comment (ankatare) [reply source/extensions/filters/network/redis_proxy/config.cc:51]: Humm :( My Bad. code pushed accidently. Corrected it.

File: test/extensions/filters/network/redis_proxy/config_test.cc (hunk 2 above)
Comment (ankatare) [reply test/extensions/filters/network/redis_proxy/config_test.cc:37]: Done. It was Local build issue

**Comment by ankatare** [comment]
@htuch Sorry . pls hold on to review it. i need to fix some local build issues.

**Comment by ankatare** [comment]
@htuch Hi, I am wondering due to some local build issues here only one file changes is visible to me however i commited code changes in 2 files. source/extensions/filters/network/redis_proxy/config.cc and test/extensions/filters/network/redis_proxy/config_test.cc

Can you also verify at your end if these 2 file changes are visibke at yur end ?

**Comment by tyxia** [comment]
> @htuch Hi, I am wondering due to some local build issues here only one file changes is visible to me however i commited code changes in 2 files. source/extensions/filters/network/redis_proxy/config.cc and test/extensions/filters/network/redis_proxy/config_test.cc
> 
> Can you also verify at your end if these 2 file changes are visibke at yur end ?

Hi @ankatare,  I can work with you here. 
From Github, i can only see one file is changed (which is source/extensions/filters/network/redis_proxy/config.cc). Did you `git add` all the changes? or was your change made in the correct git branch? You can also `git status` to check it.

Let me know if you have any questions.

**Comment by tyxia** [comment]
> > @htuch Hi, I am wondering due to some local build issues here only one file changes is visible to me however i commited code changes in 2 files. source/extensions/filters/network/redis_proxy/config.cc and test/extensions/filters/network/redis_proxy/config_test.cc
> > Can you also verify at your end if these 2 file changes are visibke at yur end ?
> 
> Hi @ankatare, I can work with you here.
> From Github, i can only see one file is changed (which is source/extensions/filters/network/redis_proxy/config.cc).
> But actually from your previous commit [2edb9e2](https://github.com/envoyproxy/envoy/commit/2edb9e276400314d9a0f42297101bdd3c0ceba78), I can see changes in both files. Maybe it has something wrong with your most recent merge main?

Hi @ankatare, I took another look. Only one file changed is expected behavior. 
The changes you made in `config_test` has already been done in #17627, so those changes in your PR disappeared once you merge main.

@htuch , I think this PR is ready to be LGed and approved

**Comment by ankatare** [comment]
@tyxia yeah.. thanks a lot for deep analysis and help. Same seems to em as well :)

**Review by htuch — APPROVED** [review]
